packages/csharp/ArmoniK.Api.Client/ResultsClientExt.cs

```diff
@@ -0,0 +1,129 @@
+    /// <param name="sessionId">The id of the session</param>
+    /// <param name="resultId">The id of the result</param>
+    /// <param name="data">The data to send</param>
+    /// <returns>
+    ///   The upload result response
+    /// </returns>
+    [PublicAPI]
+    public static async Task<UploadResultDataResponse> UploadResultData(this Results.ResultsClient client,
+                                                                        string                     sessionId,
+                                                                        string                     resultId,
+                                                                        byte[]                     data)
+    {
+      var configuration = await client.GetServiceConfigurationAsync(new Empty());
+
+      var stream = client.UploadResultData();
+
+      await stream.RequestStream.WriteAsync(new UploadResultDataRequest
+                                            {
+                                              Id = new UploadResultDataRequest.Types.ResultIdentifier
+                                                   {
+                                                     ResultId  = resultId,
+                                                     SessionId = sessionId,
+                                                   },
+                                            })
+                  .ConfigureAwait(false);
+
+      var start = 0;
+      while (start < data.Length)
+      {
+        var chunkSize = Math.Min(configuration.DataChunkMaxSize,
+                                 data.Length - start);
+
+        await stream.RequestStream.WriteAsync(new UploadResultDataRequest
+                                              {
+                                                DataChunk = UnsafeByteOperations.UnsafeWrap(data.AsMemory()
+                                                                                                .Slice(start,
+                                                                                                       chunkSize)),
+                                              })
+                    .ConfigureAwait(false);
+
+        start += chunkSize;
+      }
+
+      await stream.RequestStream.CompleteAsync()
+                  .ConfigureAwait(false);
+
+      return await stream.ResponseAsync.ConfigureAwait(false);
+    }
+
+    /// <summary>
+    ///   Download a result
+    /// </summary>
+    /// <param name="client">gRPC result client</param>
+    /// <param name="sessionId">The id of the session</param>
+    /// <param name="resultId">The id of the result</param>
+    /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+    /// <returns>
+    ///   A byte array containing the data associated to the result
+    /// </returns>
+    [PublicAPI]
+    public static async Task<byte[]> DownloadResultData(this Results.ResultsClient client,
```

Comment:
Same here. Returning a `byte[]` or a `Memory`

Reply:
an effort for #39